src/mpi/debugger/dll_mpich.c

```diff
@@ -88,6 +88,10 @@ typedef struct communicator_t {
                                  * matchine */
     int present;
     mqs_communicator comm_info; /* Info needed at the higher level */
+#ifdef HAVE_CH4_DEBUGGER_SUPPORT
+    mqs_taddr_t posted_base;    /* CH4 has the option for per-communicator queue */
```

Comment:
Do debuggers support per-comm queues?

Reply:
Yes, it is set is ch4 level:
```
#ifdef HAVE_DEBUGGER_SUPPORT
#ifndef MPIDI_CH4U_USE_PER_COMM_QUEUE
    MPIDIG_COMM(comm, posted_head_ptr) = &(MPIDI_global.posted_list);
    MPIDIG_COMM(comm, unexp_head_ptr) = &(MPIDI_global.unexp_list);
#else
    MPIDIG_COMM(comm, posted_head_ptr) = &(MPIDIG_COMM(comm, posted_list));
    MPIDIG_COMM(comm, unexp_head_ptr) = &(MPIDIG_COMM(comm, unexp_list));
#endif
#endif

```

Reply:
ch4_comm.c

Reply:
That's not my question. Does a debugger support viewing per-comm queues? Totalview, DDT, etc. Does exposing this in the comm struct result in something that the user can view?

Reply:
I tested with Totalview in 2017, it did see per-comm queue. Our license expired for now, so I cannot test with it again.

Reply:
OK, that sounds good. I think someone had reached out to me after your PR was submitted asking if that functionality existed. I'll take your word for it :).